flow/connectors/mysql/validate.go

```diff
@@ -1,14 +1,172 @@
 
+	"github.com/PeerDB-io/peerdb/flow/connectors/utils"
 	"github.com/PeerDB-io/peerdb/flow/generated/protos"
+	"github.com/PeerDB-io/peerdb/flow/shared"
 )
 
+type MySQLConnector struct {
+	conn *sql.DB
+	config *MySQLConfig
+}
+
+func (c *MySQLConnector) CheckSourceTables(ctx context.Context, tableNames []*utils.SchemaTable) error {
+	if c.conn == nil {
+		return errors.New("check tables: conn is nil")
+	}
+
+	for _, parsedTable := range tableNames {
+		query := fmt.Sprintf("SELECT 1 FROM `%s`.`%s` LIMIT 1", parsedTable.Schema, parsedTable.Table)
+		_, err := c.conn.QueryContext(ctx, query)
+		if err != nil {
+			return fmt.Errorf("error checking table %s.%s: %v", parsedTable.Schema, parsedTable.Table, err)
+		}
+	}
+	return nil
+}
+
+func (c *MySQLConnector) CheckReplicationPermissions(ctx context.Context) error {
+	if c.conn == nil {
+		return errors.New("check replication permissions: conn is nil")
+	}
+
+	var replicationPrivilege string
+	err := c.conn.QueryRowContext(ctx, "SHOW GRANTS FOR CURRENT_USER()").Scan(&replicationPrivilege)
+	if err != nil {
+		return fmt.Errorf("failed to check replication privileges: %v", err)
+	}
+
+	if !strings.Contains(replicationPrivilege, "REPLICATION SLAVE") && !strings.Contains(replicationPrivilege, "REPLICATION CLIENT") {
+		return errors.New("MySQL user does not have replication privileges")
+	}
+
+	return nil
+}
+
+func (c *MySQLConnector) CheckReplicationConnectivity(ctx context.Context) error {
+	if c.conn == nil {
+		return errors.New("check replication connectivity: conn is nil")
+	}
+
+	var masterLogFile string
+	var masterLogPos int
+
+	err := c.conn.QueryRowContext(ctx, "SHOW MASTER STATUS").Scan(&masterLogFile, &masterLogPos)
+	if err != nil {
+		// Handle case where SHOW MASTER STATUS returns no rows (binary logging disabled)
+		if errors.Is(err, sql.ErrNoRows) {
+			return errors.New("binary logging is disabled on this MySQL server")
+		}
+		return fmt.Errorf("failed to check replication status: %v", err)
+	}
+
+	// Additional validation: Check if the values are valid
+	if masterLogFile == "" || masterLogPos <= 0 {
+		return errors.New("invalid replication status: missing log file or position")
+	}
+
+	return nil
+}
+
+func (c *MySQLConnector) CheckBinlogSettings(ctx context.Context) error {
+	if c.conn == nil {
+		return errors.New("check binlog settings: conn is nil")
+	}
+
+	// Check binlog_expire_logs_seconds
+	var expireSeconds int
+	err := c.conn.QueryRowContext(ctx, "SELECT @@binlog_expire_logs_seconds").Scan(&expireSeconds)
+	if err != nil {
+		return fmt.Errorf("failed to retrieve binlog_expire_logs_seconds: %v", err)
+	}
+	if expireSeconds <= 86400 {
+		return errors.New("binlog_expire_logs_seconds is too low. Must be greater than 1 day")
+	}
+
+	// Check binlog_format
+	var binlogFormat string
+	err = c.conn.QueryRowContext(ctx, "SELECT @@binlog_format").Scan(&binlogFormat)
+	if err != nil {
+		return fmt.Errorf("failed to retrieve binlog_format: %v", err)
```

Comment:
```suggestion
	if err := c.conn.QueryRowContext(ctx, "SELECT @@binlog_format").Scan(&binlogFormat); err != nil {
		return fmt.Errorf("failed to retrieve binlog_format: %w", err)
```
been trying to have consistent error scoping, this decision has avoided a couple issues

%w over %v in `fmt.Errorf` allows `errors.As` / `errors.Is` to pick up the underlying error, which is important for us to get mysql error codes